packages/@react-native-windows/cli/src/runWindows/utils/commandWithProgress.ts

```diff
@@ -154,3 +159,39 @@ export function newSuccess(text: string) {
 export function newInfo(text: string) {
   newSpinner(text).info(text);
 }
+
+/**
+ * Sets the process exit code and offers some information at the end of a CLI command.
+ * @param loggingIsEnabled Is verbose logging enabled.
+ * @param error The error caught during the process, if any.
+ */
+export function setExitProcessWithError(
+  loggingIsEnabled?: boolean,
+  error?: Error,
+): void {
+  if (error) {
+    const errorType =
+      error instanceof CodedError ? (error as CodedError).type : 'Unknown';
+
+    process.exitCode = CodedErrors[errorType];
+
+    if (loggingIsEnabled) {
+      console.log(
+        `Command failed with error ${chalk.bold(errorType)}: ${error.message}`,
+      );
+      if (Telemetry.isEnabled()) {
+        console.log(
+          `Your telemetry sessionId was ${chalk.bold(
+            Telemetry.getSessionId(),
+          )}`,
+        );
+      }
+    } else {
+      console.log(
+        `Command failed. Re-run the command with ${chalk.bold(
+          '--logging',
+        )} for more information.`,
+      );
+    }
+  }
```

Comment:
Does this function do nothing if no Error is passed in? Why is Error optional then?